### PR TITLE
feat(frontend/backend): add support for Exchange Web Services (EWS)

### PR DESCRIPTION
**Pull request by kroky**
Feature request: https://github.com/cypht-org/cypht/issues/247

Work in progress:
1. Add Mailbox bridge for imap and similar modules to work with protocols other than IMAP - e.g. EWS (WIP).
2. Implement EWS methods using php-ews package and raw SOAP calls.
3. Review and write tests.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -55,7 +55,8 @@
         "twbs/bootstrap": "^5.3",
         "twbs/bootstrap-icons": "^1.11",
         "webklex/composer-info": "^0.0.1",
-        "zbateson/mail-mime-parser": "^2.4"
+        "zbateson/mail-mime-parser": "^2.4",
+        "garethp/php-ews": "^0.10.1"
```
Comment (marclaporte) [inline composer.json:59]: alphabetical would be better.

**Comment by marclaporte** [comment]
Email clients come in many varieties:

- Open Source or proprietary
- Desktop, webmail or mobile client
- etc

Cypht Webmail will be soon one of the rare mail clients in the world to support IMAP/SMTP, JMAP and EWS!

File: composer.json (hunk 1 above)
Comment (kroky) [reply composer.json]: Done. Also switched to latest master upstream. We need this to be merged before using the full set of features developed in this MR: https://github.com/Garethp/php-ews/pull/266

**Comment by kroky** [comment]
This is no longer a draft and can be more extensively tested. Ideas how? I am able to create exchange accounts in our test exchange system.

**Comment by marclaporte** [comment]
@kambereBr and @Shadow243 are now testing.

**Comment by marclaporte** [comment]
General info: https://github.com/Garethp/php-ews/issues/267

**Comment by marclaporte** [comment]
@kroky "This branch has conflicts that must be resolved"

Also, can test site have latest code from master and PR?

Thanks!

**Comment by Neustradamus** [comment]
To follow this PR ^^

**Comment by marclaporte** [comment]
For the record: @Shadow243 and @kroky are coordinating on testing. Expect some work in the coming 10 days and then, we should be good to merge to master, and unleash it onto the world :-)

**Comment by marclaporte** [comment]
https://github.com/Garethp/php-ews/pull/266 has been merged

**Comment by Shadow243** [comment]
> Updated to latest php-ews upstream client which has the latest wsdl file changes accepted, so now ArchiveItem is also there to use.
> 
> 
> 
> Merged latest master and resolved the conflicts. Things look good on my end. I think this is ready to be merged unless testers have something else to report? 

I’ll go over it again to confirm.

**Comment by kroky** [comment]
We can fix in master from now on. I will merge as this gets too many conflicts from other features.

**Comment by marclaporte** [comment]
Related: [Feature] Add support for Microsoft Graph, which replaces Microsoft Exchange Web Services (EWS): https://github.com/cypht-org/cypht/issues/1659

**Pull request merged**